### PR TITLE
[newrelic-pixie] [newrelic-k8s-metrics-adapter] Add image pull secrets (#597)

### LINKED ISSUE #597: [all] require pull secrets as part of helm charts
**Is your feature request related to a problem? Please describe.**

It feels like there is an endless stream of images being added to the charts with no way to specify pull secrets. It makes operating in certain environments incredibly difficult to nearly impossible due to the Dockerhub rate limitations.

Most recent example: https://github.com/newrelic/helm-charts/blob/master/charts/newrelic-infra-operator/values.yaml#L89-L117

After updating the chart version I can now no longer install nri-bundle due to this.

**Describe the solution you'd like**

Reject additions to Helm charts that do not include pull secrets as part of the template.

**Describe alternatives you've considered**

There are no alternatives

### PR DESCRIPTION
**Pull request by viniciusfs**
#### Is this a new chart
No.

#### What this PR does / why we need it:
Add imagePullSecrets on deployments/jobs where it is missing.

#### Which issue this PR fixes
  - fixes #597 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[mychartname]`)

### CODE REVIEW

File: charts/newrelic-pixie/templates/deployment.yaml [hunk 1]
```diff
@@ -18,6 +18,10 @@ spec:
         app.kubernetes.io/name: {{ template "newrelic-pixie.name" . }}
         release: {{.Release.Name }}
     spec:
+      {{- if .Values.image.pullSecrets }}
+      imagePullSecrets:
+{{ toYaml .Values.image.pullSecrets | indent 8 }}
```
Comment (kang-makes) [inline charts/newrelic-pixie/templates/deployment.yaml:23]:
```suggestion
      imagePullSecrets:
        {{- toYaml .Values.image.pullSecrets | nindent 8 }}
```

File: charts/newrelic-k8s-metrics-adapter/templates/apiservice/job-patch/job-patchAPIService.yaml [hunk 2]
```diff
@@ -18,6 +18,10 @@ spec:
         app: {{ template "newrelic-k8s-metrics-adapter.name" $ }}
 {{ include "newrelic-k8s-metrics-adapter.labels" $ | indent 8 }}
     spec:
+      {{- if .Values.apiServicePatchJob.image.pullSecrets }}
+      imagePullSecrets:
+{{ toYaml .Values.apiServicePatchJob.image.pullSecrets | indent 8 }}
```
Comment (kang-makes) [inline charts/newrelic-k8s-metrics-adapter/templates/apiservice/job-patch/job-patchAPIService.yaml:23]:
```suggestion
      imagePullSecrets:
        {{- toYaml .Values.apiServicePatchJob.image.pullSecrets | nindent 8 }}
```

**Review by kang-makes — COMMENTED** [review]
Added to the Roberto's comments I would add like a `.global.pullSecrets` or something like that so we don't have to do it in all the charts individually.

**Comment by roobre** [comment]
> Added to the Roberto's comments I would add like a `.global.pullSecrets` or something like that so we don't have to do it in all the charts individually.

I definitely agree but I'd suggest to leave that for a follow-up PR when we add it to all of our charts rather than just one :)

**Comment by viniciusfs** [comment]
@roobre done! :)

File: charts/newrelic-pixie/values.yaml [hunk 3]
```diff
@@ -21,6 +21,8 @@ image:
   repository: newrelic/newrelic-pixie-integration
   tag: ""
   pullPolicy: IfNotPresent
+  # pullSecrets:
+  # - name: regsecret
```
Comment (kang-makes) [inline charts/newrelic-pixie/values.yaml:25]: Take a look at variables like these:
```yaml
proxy: {}
nodeSelector: {}
tolerations: []
affinity: {}
```
[Helm falsiness](https://helm.sh/docs/chart_template_guide/control_structures/#ifelse) is going to detect if as false if is empty. Better empty than commented :)

The comment as an example is perfect.
```suggestion
  pullSecrets: []
  # - name: regsecret
```

**Comment by viniciusfs** [comment]
Added a default empty list as suggested. Sorry about delay, I was sick for few days.

**Comment by kang-makes** [comment]
No prob, there are conflict because the other charts have been upgraded. Rebase or bump the version again and I will merge it as soon as possible to not stale this PR.

I don't want to abuse your patience anymore :)

**Comment by roobre** [comment]
Hey @viniciusfs, this is looking good from our side, if you can rebase the changes we're ready to get it merged

**Review by roobre — APPROVED** [review]

**Pull request merged**